core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -684,7 +713,7 @@ private[spark] class BlockManager(
           // take a significant amount of time. To get rid of these stale entries
           // we refresh the block locations after a certain number of fetch failures
           if (runningFailureCount >= maxFailuresBeforeLocationRefresh) {
-            locationIterator = getLocations(blockId).iterator
+            locationIterator = sortLocations(master.getLocationsAndStatus(blockId)._1).iterator
```

Comment:
It looks `getLocations` is still useful for this LOC.

Reply:
Agreed, will change it.